lib/src/tests/MGPOptional.spec.ts

```diff
@@ -123,4 +123,16 @@ describe('MGPOptional', () => {
             expect(optional.map(addOne)).toEqual(MGPOptional.of(42));
         });
     });
+
+    it('should be able to get on an else branch of isAbsent', () => {
+        function f(v: MGPOptional<number>): number {
+            if (v.isAbsent()) {
+                return 0;
+            } else {
+                return v.get();
+            }
+        }
+        expect(f(MGPOptional.of(5))).toBe(5);
```

Comment:
c'est pas de bon sang tso, tu test la fonction f, pas notre code

Reply:
on croirait hein ? Mais non, ce code teste bien la lib: je peux faire une modification à la lib qui casse la compilation de ce test, mais qui - sans ce test - permettrait à lib/ de compiler, et casserait la compil de MGP.
Par soucis de self-contenance, ce code vient donc "tester qu'on peut utiliser lib comme il faut"

Reply:
Mais euh, c'est pas mieux de faire a = MGPOptional.of(5) 
expect(a.isAbsent()).toBeFalse()
expect(a.get()).toBe(5) ?
C'est la même chose en plus clair

Reply:
nope, le bug se situe dans le mix entre isAbsent et le typage des if en typescript, il n'apparaît pas si on a pas un if sur isAbsent + un .get() dans le else : 
  - si isAbsent est typé par "this is MGPOptionalAbsent", typescript cast la variable en "MGPOptionalAbsent" dans le then
  - par contre, il cast la variable en "never" dans le else !!! Et .get() n'est pas défini sur never

Reply:
ouais